### PR TITLE
HOT templates for Moodle

### PR DESCRIPTION
**Pull request by cristiano-sala**
My first shot at orchestrating the setup of Moodle on the Catalyst Cloud. Contains YAML template and env files, bash scripts for base and application software provisioning and configuration.

### CODE REVIEW

File: Moodle_2servers/Moodle_2servers.yaml [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+parameters:
+
+# Infrastructure
+
+  key_name:
+    type: string
+    description: Name of an existing key pair to use for the server
+    constraints:
+      - custom_constraint: nova.keypair
+    default: cristiano
```
Comment (xavpaice) [inline Moodle_2servers/Moodle_2servers.yaml]: You might want to anonymize this a bit more?

Comment (cristiano-sala) [inline Moodle_2servers/Moodle_2servers.yaml]: Yep, thanks for spotting it. I thought I had removed it everywhere, but I was wrong

File: Moodle_2servers/Moodle_2servers.yaml [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    type: string
+    description: >
+      ID or name of public network for which floating IP addresses will be allocated
+    default: public-net
+
+  public_net_id:
+    type: string
+    description: >
+      ID of public network for which floating IP addresses will be allocated
+    default: 849ab1e9-7ac5-4618-8801-e6176fbbcf30
```
Comment (xavpaice) [inline Moodle_2servers/Moodle_2servers.yaml]: might be reasonable to have no default, and put this in the environment file too?

File: Moodle_2servers/Moodle_2servers.yaml [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+  private_net_pool_end:
+    type: string
+    description: End of private network IP address allocation pool
+    default: 192.168.0.250
+
+  private_net_dns_servers:
+    type: comma_delimited_list
+    description: DNS servers for the private network
+    default: [ 202.78.247.197, 202.78.247.198, 202.78.247.199 ]
```
Comment (xavpaice) [inline Moodle_2servers/Moodle_2servers.yaml]: These are specific to a region so same thing, in the environment file?

Comment (cristiano-sala) [inline Moodle_2servers/Moodle_2servers.yaml]: It probably makes sense to move a lot of regional/local info to the environment file, my initial concern was that it makes it less usable out-of-the-box for other Catalyst users. In any case, good point

**Comment by cristiano-sala** [comment]
I've done some overall cleanup and further anonymisation. Let me know what you think

**Comment by cristiano-sala** [comment]
Thanks!

**Pull request merged**